src/buffer/out/OutputCellIterator.cpp

```diff
@@ -206,6 +206,40 @@ OutputCellIterator& OutputCellIterator::operator++()
     // Keep track of total distance moved (cells filled)
     _distance++;
 
+    if (_mode == Mode::Loose && _currentView.DbcsAttr().IsSingle())

```

Comment:
This branch here needs to be highly compact & optimized to avoid potential "I-Cache spill" (similar to the one in #10621), so I only handle the condition when `_currentView.DbcsAttr().IsSingle()`